actions/aws-upload.js

```diff
@@ -0,0 +1,196 @@
+      gzip: true,
+      file: path.resolve(toPath),
+    };
+
+    console.log('Tarring directory:', process.cwd());
+    console.log('Writing tar to:', toPath);
+
+    // Tar the file
+    await tar.c(params, ['.']);
+    // Return to original working directory
+    process.chdir(cwd);
+  }
+
+  catch(err) {
+    process.chdir(directory);
+    throw err;
+  }
+}
+
+/**
+ * options {
+ *   directory: The directory who's contents we upload
+ *   tarPath: The path where the tar file should be saved. Default is '.'
+ *   bucketPath: The path for the bucket and file path to upload to in AWS <bucket name>/<file path>
+ *   awsFileName: The name of the file to use in the AWS system
+ *   makePublic: Set to true if the file is supposed to be public in the AWS repo
+ * }
+ * This uploads a directory to AWS using the aws-sdk by tarring the contents of the directory then using
+ * the sdk to upload the file.Normal rules apply for the awk-sdk authentication.
+ *
+ * The recommended authentication method for this method is to use the environment variables.
+ * AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY
+ *
+ * This supports taking the file path to upload as a parameter or from the environment var:
+ * AWS_UPLOAD
+ *
+ * This is the path
+ * AWS_TAR_PATH
+ *
+ * This supports taking the path for the bucket to upload to via parameter bucketPath or env:
+ * AWS_BUCKET
+ *
+ * @returns An error message if the upload failed. Nothing if successful.
+ */
+async function awsTarUpload(options = {}) {
+  const {
+    // When this is defined with valid content, autorelease will attempt to upload a tar of the
+    // distribution with the provided credentials. This should be populated in the following fashion:
+    // <bucket name>/<path to file>
+    AWS_BUCKET,
+    // This should be provided to indicate which directory to upload
+    AWS_UPLOAD,
+    // The path of the tar file's save location
+    AWS_TAR_PATH,
+    // When set, makes the uploaded file public
+    AWS_PUBLIC,
+  } = process.env;
+
+  const {
+    awsFileName = '',
+    bucketPath = AWS_BUCKET || '',
+    directory = AWS_UPLOAD || '',
+    makePublic = Boolean(AWS_PUBLIC),
+    tarPath = AWS_TAR_PATH || '',
+  } = options;
+
+  // Split the path so we can use the start as the bucket name and the rest as the file pathing
+  const bucketPathing = bucketPath.split('\/');
+  // This is the selected path for uploading to the system
+  const pathToUpload = directory || AWS_UPLOAD;
+
+  // Bucket names must be unique across all S3 users
+  const myBucket = bucketPathing[0];
+
+  if (awsFileName) {
+    // Add the file name as a part of the pathing
+    bucketPathing.push(awsFileName);
+  }
+
+  const myKey = bucketPathing.slice(1).join('/');
+
+  // Validate all inputs for the process
+  if (!bucketPath || !bucketPathing || !pathToUpload) {
+    reject(new Error(`All needed inputs for the AWS upload were not provided or invalid. bucket: ${myBucket}, bucketPath: ${myKey}, toUpload: ${pathToUpload}`));
```

Comment:
What is `reject` at this point? I don't see it. I think you mean `throw`

Reply:
Ah missed that one, converted from Promises 👍 will dew